### PR TITLE
Merging with main repo

### PR DESCRIPTION
**Pull request by pdolega**
This PR summarizes couple of things we did on our side ( Nexelem's fork). To list them all:
- updated libraries from titan to newest 2.5.x versions
- made compiler fail for 2.11 if there are any warnings
- updated SBT to latest / greatest version 0.13.7
- added handling of Option values (Options in case classes mapped to vertices)
- added some tests for above feature
- removed all the compiler warnings in compilation for 2.11

Also - made 2.11 the default one (instead of 2.10 - still 2.10 is available).

### CODE REVIEW

**Comment by anvie** [comment]
Hi @pdolega thx for the PR, but currently I have no available time to doing any programming stuff like for testing using Scala 2.11 etc, will check it soon when I have a time.

**Comment by pdolega** [comment]
Hi @anvie 

Sure - no probs. Please do whenever you have time available, at your convenience.
You've already did a helluva good work with this library. It proved to be pretty useful for us. Thanks !

File: build.sbt [hunk 1]
```diff
@@ -26,36 +26,49 @@ libraryDependencies ++= Seq(
...
 profileName := "com.ansvia"
 
 publishTo <<= version { (v:String) =>
-      val nexus = "https://oss.sonatype.org"
-      if(v.trim.endsWith("SNAPSHOT") || """.+\-\d{8}+$""".r.pattern.matcher(v.trim).matches())
-          Some("snapshots" at nexus + "/content/repositories/snapshots")
-      else
-	  Some("releases"  at nexus + "/service/local/staging/deploy/maven2")
-  }
+  val nexus = "https://oss.sonatype.org"
```
Comment (anvie) [inline build.sbt]: I'm using 4 spaces as standard indentation, it is okay if I convert all 2 spaces into 4 spaces indentation?

**Comment by anvie** [comment]
If you don't mind, please add your name into the `<developer>` section :) https://github.com/anvie/blueprints-scala/blob/2b9e0c6e851861614fdb1c52e61f95f74b1ec960/build.sbt#L95

**Pull request merged**